File: LeetCode/1389. Create Target Array in the Given Order/solution.cpp
/**
 * Runtime: 0 ms
 * Memory Usage: 8.2 MB 
 */
class Solution {
public:
    vector<int> createTargetArray(vector<int>& nums, vector<int>& index) {
        vector<int> ans;
        for (int i = 0; i < nums.size(); ++ i) {
            ans.insert(ans.begin() + index[i], nums[i]);
        }
        return ans;
    }
};

File: LeetCode/1389. Create Target Array in the Given Order/solution.py
"""
Runtime: 32 ms
Memory Usage: 14.9 MB
"""
class Solution:
    def createTargetArray(self, nums: List[int], index: List[int]) -> List[int]:
        ans = []
        for i, num in enumerate(nums):
            ans.insert(index[i], num)
        return ans
